lib/handler/connect.c

```diff
@@ -533,8 +553,14 @@ static void tcp_on_read(h2o_socket_t *_sock, const char *err)
 
     if (err == NULL) {
         h2o_iovec_t vec = h2o_iovec_init(self->sock->input->bytes, self->sock->input->size);
+        H2O_PROBE_CONN(CONNECT_TCP_READ, self->src_req->conn, self->src_req->req_index, vec.len);
         h2o_send(self->src_req, &vec, 1, H2O_SEND_STATE_IN_PROGRESS);
     } else {
+        if (err == h2o_socket_error_closed) {
+            H2O_PROBE_CONN(CONNECT_TCP_READ_CLOSED, self->src_req->conn, self->src_req->req_index);
+        } else {
+            H2O_PROBE_CONN(CONNECT_TCP_READ_ERROR, self->src_req->conn, self->src_req->req_index, err);
```

Comment:
Am I correct to understand that the intent for having two probes that show closure is to have the possibility of just monitoring abnormal behavior, rather than monitoring all close events?

If so that's fine (but otherwise I would prefer merging the two).

Reply:
I was thinking that it is useful to know whether it was the client or the origin who decided to close the connection, that is why there is write-closed probe and a read-closed probe.

I do not have a good reason for separating read-closed from read-error.  I will see if I can make it work sensibly with only one probe here.

Reply:
Probe `connect-tcp-read-closed` removed in 21c8f1dfa.

Reply:
It is now one probe since cd07df4.